src/Overlay/Modal/Modal.styled.ts

```diff
@@ -0,0 +1,82 @@
+import styled, { css, keyframes } from 'styled-components'
+import { Button } from '../../Button'
+import { titleLarge } from '../../theme'
+
+
+const fadeInAnimation = keyframes`
+  0% {
+    opacity: 0;
+    transform: scale(0);
+  }
+  100% {
+    opacity: 1;
+    transform: scale(1);
+  }
+`
+
+export const Dialog = styled.dialog`
+    background-color: var(--md-sys-color-surface-container);
+    border: none;
+    border-radius: var(--border-radius-m);
+    flex-direction: column;
+    gap: 16px;
+    padding: 0;
+    min-width: 200px;
+    min-height: 100px;
+    max-width: 85%;
+    max-height: 85%;
+
+    /* Backdrop property affects inactive area around modal */
+    &::backdrop {
+        background-color: rgba(0, 0, 0, 0.3);
+    }
+
+    /* Styles for dialogs that carry modal behavior */
+    &:modal {
+    }
+
+    &:modal[open] {
+      animation: ${fadeInAnimation} 0.25s ease-in-out forwards, ${fadeInAnimation} 0.25s ease-in-out backwards;
```

Comment:
Not sure why there is two animations here, one forward and one backward?

Reply:
Yes, that was the aim. Problem is, only opening animation executes. I have seen the issue being noticed, there are some workarounds like implementing delay...